examples/2d/elixir_advection_callbacks.jl

```diff
@@ -0,0 +1,154 @@
+
+  min_val, max_val = extrema(u_ode)
+  push!(example_callback.times, t)
+  push!(example_callback.min_values, min_val)
+  push!(example_callback.max_values, max_val)
+
+  # avoid re-evaluating possible FSAL stages
+  u_modified!(integrator, false)
+  return nothing
+end
+
+# This method is used to wrap an `ExampleCallback` inside a `DiscreteCallback`
+# which gets called after every RK step. You can pass an additional initialization
+# method and a separate condition specifying whether the callback shall be called.
+function ExampleCallback(; message::String)
+  # Call the `ExampleCallback` after every RK step.
+  condition = (u_ode, t, integrator) -> true
+
+  # You can optionally pass an initialization method. There, you can access the
+  # `ExampleCallback` as `cb.affect!`.
+  initialize = (cb, u_ode, t, integrator) -> println(cb.affect!.message)
+
+  example_callback = ExampleCallback(message)
+
+  DiscreteCallback(condition, example_callback,
+                   save_positions=(false,false),
+                   initialize=initialize)
+end
+
+end # module TrixiExtensionExample
+
+import .TrixiExtensionExample
+
+###############################################################################
+# semidiscretization of the linear advection equation
+
+advectionvelocity = (1.0, 1.0)
+# advectionvelocity = (0.2, -0.3)
+equations = LinearScalarAdvectionEquation2D(advectionvelocity)
+
+initial_condition = initial_condition_convergence_test
+
+surface_flux = flux_lax_friedrichs
+solver = DGSEM(3, surface_flux)
+
+coordinates_min = (-1, -1)
+coordinates_max = ( 1,  1)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=4,
+                n_cells_max=30_000)
+
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan);
+
+summary_callback = SummaryCallback()
+
+stepsize_callback = StepsizeCallback(cfl=1.6)
+
+save_solution = SaveSolutionCallback(interval=100,
+                                     save_initial_solution=true,
+                                     save_final_solution=true,
+                                     solution_variables=:conservative)
+
+example_callback = TrixiExtensionExample.ExampleCallback(message="안녕하세요?")
+
+analysis_interval = 100
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval,
+                                     extra_analysis_integrals=(entropy, energy_total))
+
+callbacks = CallbackSet(summary_callback, stepsize_callback,
+                        save_solution,
+                        example_callback,
+                        analysis_callback, alive_callback)
+
+example_limiter! = TrixiExtensionExample.ExampleLimiter()
+stage_limiter! = example_limiter!
+step_limiter!  = example_limiter!
```

Comment:
Maybe a comment here (or with `solve` below) about what the difference between stage and step limiter is and why both need to be set to the same value here would be good.

Reply:
Added in 8f2fe48da94f56682ccfd55b2a8bdec0e6052337.

Reply:
Thanks, good comment